dune_client/api/execution.py

```diff
@@ -115,18 +120,22 @@ def get_execution_results_csv(
         use this method for large results where you want lower CPU and memory overhead
         if you need metadata information use get_results() or get_status()
         """
-        params = self._build_parameters(
-            columns=columns,
-            sample_count=sample_count,
-            filters=filters,
-            sort_by=sort_by,
-            limit=limit,
-            offset=offset,
-        )
+        build_params = None
+        if params is not None:
+            build_params = self._build_parameters(
+                params={
+                    "columns": params.columns,
+                    "sample_count": params.sample_count,
+                    "filters": params.filters,
+                    "sort_by": params.sort_by,
+                    "limit": params.limit,
+                    "offset": params.offset,
+                }
+            )
```

Comment:
I think this is unnecessary. and can either be passed to the next function as is. It should only be at the very last step it needs to be transformed to a dict.